workflows/prognostic_run_diags/fv3net/diagnostics/prognostic_run/metrics.py

```diff
@@ -138,6 +138,25 @@ def rmse_time_mean(diags):
     return rms_of_time_mean_bias
 
 
+for percentile in [25, 50, 75, 90, 99, 99.9]:
+
+    @add_to_metrics(f"percentile_{percentile}")
+    def percentile_metric(diags, p=percentile):
+        histogram = grab_diag(diags, "histogram")
+        percentiles = xr.Dataset()
+        data_vars = [v for v in histogram.data_vars if not v.endswith("bin_width")]
+        for varname in data_vars:
+            bins = histogram[f"{varname}_bins"].values
+            bin_width = histogram[f"{varname}_bin_width"]
+            bin_midpoints = bins + 0.5 * bin_width
+            cumulative_distribution = np.cumsum(histogram[varname][:-1] * bin_width)
+            percentiles[varname] = bin_midpoints[
+                np.argmin(np.abs(cumulative_distribution.values - p / 100))
+            ]
+        restore_units(histogram, percentiles)
```

Comment:
"restore"? Sounds like we use a lot of verbs for this operation of grabbing metadata from one xarray object and applying it to another. I think "assign" is more widely used. E.g.
```
assign_units(from_=histogram, to=percentiles)
```

Reply:
okay.. but it is an existing function so I think I'll just leave it.